components/datadog/dockeragentparams/params.go

```diff
@@ -223,3 +233,11 @@ func WithExtraComposeManifest(name string, content pulumi.StringInput) func(*Par
 		return nil
 	}
 }
+
+// WithExtraComposeInlineManifest adds extra docker.ComposeInlineManifest
+func WithExtraComposeInlineManifest(cpms ...docker.ComposeInlineManifest) func(*Params) error {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
What is the advantage of this compared to the existing `WithExtraComposeManifest`?

Reply:
I'm creating the ComposeInlineManifest [here](https://github.com/DataDog/test-infra-definitions/pull/1248/files#diff-58bafaac01a0b66b758bd097fe8ddc952f505e1dfdba02b1bc273502289713a8R25) so would like to be able to just do:
```
suiteParams := []e2e.SuiteOption{e2e.WithProvisioner(
		awsdocker.Provisioner(
			awsdocker.WithAgentOptions(
				dockeragentparams.WithLogs(),
				dockeragentparams.WithJMX(),
				dockeragentparams.WithExtraComposeInlineManifest(
					jmxfetch.DockerComposeManifest,
					jmxfetch.DockerComposeAllMetricsManifest),
			)))}
```

Before in my code it looked more like:
```
		awsdocker.Provisioner(
			awsdocker.WithAgentOptions(
				dockeragentparams.WithLogs(),
				dockeragentparams.WithJMX(),
				dockeragentparams.WithExtraComposeManifest("jmx-test-app", jmxfetch.DockerComposeManifest.Content),
				dockeragentparams.WithExtraComposeManifest("jmx-test-app-labels", jmxfetch.DockerComposeAllMetricsManifest.Content),
			)))}

```
And was never really using the `Name` from `docker.ComposeInlineManifest`. I looked at what `WithExtraComposeManifest` was doing and it was just creating the same things I was creating [here](https://github.com/DataDog/test-infra-definitions/pull/1248/files#diff-58bafaac01a0b66b758bd097fe8ddc952f505e1dfdba02b1bc273502289713a8R25).

Reply:
Oh I like this actually, it is closer to the ECS and K8S apps definitions